pytensor/tensor/subtensor.py

```diff
@@ -162,13 +187,21 @@ def as_index_literal(
     ------
     NotScalarConstantError
     """
-    if idx == np.newaxis or isinstance(getattr(idx, "type", None), NoneTypeT):
-        return np.newaxis
+    if isinstance(idx, int | np.integer):
+        return idx
```

Comment:
This return simplifies the graph produced by `get_canonical_form_slice`, because previously `int` inputs resulted in the `NotScalarConstantError` and going down into the complicated parts of `get_canonical_form_slice`.